huaweicloud/services/vpc/data_source_huaweicloud_networking_secgroups.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+func DataSourceNetworkingSecGroups() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceNetworkingSecGroupsRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"security_groups": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"enterprise_project_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"created_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"updated_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getSecGroupDetail(secGroup groups.SecurityGroup) map[string]interface{} {
+	return map[string]interface{}{
+		"id":                    secGroup.ID,
+		"name":                  secGroup.Name,
+		"enterprise_project_id": secGroup.EnterpriseProjectId,
+		"description":           secGroup.Description,
+		"created_at":            secGroup.CreatedAt,
+		"updated_at":            secGroup.UpdatedAt,
+	}
+}
+
+func filterAvailableSecGroups(d *schema.ResourceData, secGroups []groups.SecurityGroup) ([]map[string]interface{},
+	[]string) {
+	secGroupCopy := secGroups
+
+	// build filter by description content.
+	if desc, ok := d.GetOk("description"); ok {
+		tmp := make([]groups.SecurityGroup, 0, len(secGroupCopy))
+		for _, secgroup := range secGroupCopy {
+			if strings.Contains(secgroup.Description, desc.(string)) {
+				tmp = append(tmp, secgroup)
+			}
+		}
+		secGroupCopy = tmp
+	}
+
+	result := make([]map[string]interface{}, len(secGroupCopy))
+	ids := make([]string, len(secGroupCopy))
+	for i, secGroup := range secGroupCopy {
+		ids[i] = secGroup.ID
+		result[i] = getSecGroupDetail(secGroup)
+	}
+
+	return result, ids
+}
+
+func dataSourceNetworkingSecGroupsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	networkingClient, err := config.NetworkingV3Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud networking client: %s", err)
+	}
+
+	listOpts := groups.ListOpts{
```

Comment:
the list API can support querying security groups by description, please add some reasons why we do this?